wolfcrypt/src/random.c

```diff
@@ -3553,20 +3567,37 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
 
 #ifndef NO_FILESYSTEM
     #ifndef NO_DEV_URANDOM /* way to disable use of /dev/urandom */
-        os->fd = open("/dev/urandom", O_RDONLY);
-        #if defined(DEBUG_WOLFSSL)
-            WOLFSSL_MSG("opened /dev/urandom.");
+        #ifdef WOLFSSL_KEEP_RNG_SEED_FD_OPEN
```

Comment:
This section might be easier to read with two separate versions of the code... one for existing and another for keeping seed file descriptor open.